src/audio/module_adapter/module_adapter.c

```diff
@@ -240,9 +241,17 @@ static int module_adapter_dp_queue_prepare(struct comp_dev *dev)
 			 &sink_buffer->stream.runtime_stream_params,
 			 sizeof(sink_buffer->stream.runtime_stream_params));
 		/* calculate time required the module to provide OBS data portion - a period */
-		unsigned int sink_period = 1000000 * sink_get_min_free_space(mod->sinks[i]) /
+		uint32_t min_free_size =
+			sink_get_frame_bytes(mod->sinks[i]) *
+			(sink_get_rate(mod->sinks[i]) / LL_TIMER_PERIOD_US);
```

Comment:
this doesn't look correct. Apparently you're trying to calculate number of bytes per period. `sink_get_rate()` is the number of frames per second, `LL_TIMER_PERIOD_US` is the number of microseconds per period. Since the latter is usually 1000, it happens to work, but what you really need is to divide the number of frames per second by the number of periods per second, i.e.
```
sink_get_rate(mod->sinks[i]) / (MICROSECONDS_PER_SECOND / LL_TIMER_PERIOD_US)
```

Reply:
fixed